pkg/controller/database/mysqlserver/managed.go

```diff
@@ -103,102 +103,98 @@ func (c *connecter) Connect(ctx context.Context, mg resource.Managed) (resource.
+	if err := e.kube.Update(ctx, cr); err != nil {
+		return resource.ExternalObservation{}, err
+	}
+	cr.Status.AtProvider = azuresql.GenerateMySQLObservation(server)
 
-	s.Status.State = string(external.UserVisibleState)
-	s.Status.ProviderID = azure.ToString(external.ID)
-	s.Status.Endpoint = azure.ToString(external.FullyQualifiedDomainName)
-
-	switch external.UserVisibleState {
-	case mysql.ServerStateReady:
-		s.SetConditions(runtimev1alpha1.Available())
-		if s.Status.Endpoint != "" {
-			resource.SetBindable(s)
-		}
+	switch cr.Status.AtProvider.UserVisibleState {
+	case azuresql.StateReady:
+		cr.SetConditions(runtimev1alpha1.Available())
+		resource.SetBindable(cr)
 	default:
-		s.SetConditions(runtimev1alpha1.Unavailable())
+		cr.SetConditions(runtimev1alpha1.Unavailable())
 	}
 
-	o := resource.ExternalObservation{
+	return resource.ExternalObservation{
 		ResourceExists:   true,
-		ResourceUpToDate: true, // NOTE(negz): We don't yet support updating Azure SQL servers.
+		ResourceUpToDate: azuresql.IsMySQLUpToDate(cr.Spec.ForProvider, server),
 		ConnectionDetails: resource.ConnectionDetails{
-			runtimev1alpha1.ResourceCredentialsSecretEndpointKey: []byte(s.Status.Endpoint),
-			runtimev1alpha1.ResourceCredentialsSecretUserKey:     []byte(fmt.Sprintf("%s@%s", s.Spec.AdminLoginName, s.GetName())),
+			runtimev1alpha1.ResourceCredentialsSecretEndpointKey: []byte(cr.Status.AtProvider.FullyQualifiedDomainName),
+			runtimev1alpha1.ResourceCredentialsSecretUserKey:     []byte(fmt.Sprintf("%s@%s", cr.Spec.ForProvider.AdministratorLogin, meta.GetExternalName(cr))),
 		},
-	}
-
-	return o, nil
+	}, nil
 }
 
 func (e *external) Create(ctx context.Context, mg resource.Managed) (resource.ExternalCreation, error) {
-	s, ok := mg.(*v1alpha3.MySQLServer)
+	cr, ok := mg.(*v1alpha3.MySQLServer)
 	if !ok {
 		return resource.ExternalCreation{}, errors.New(errNotMySQLServer)
 	}
 
-	s.SetConditions(runtimev1alpha1.Creating())
-
+	cr.SetConditions(runtimev1alpha1.Creating())
 	pw, err := e.newPasswordFn(passwordDataLen)
 	if err != nil {
 		return resource.ExternalCreation{}, errors.Wrap(err, errGenPassword)
 	}
-
-	if err := e.client.CreateServer(ctx, s, pw); err != nil {
+	if err := e.client.CreateServer(ctx, cr, pw); err != nil {
 		return resource.ExternalCreation{}, errors.Wrap(err, errCreateMySQLServer)
 	}
 
-	ec := resource.ExternalCreation{
+	return resource.ExternalCreation{
 		ConnectionDetails: resource.ConnectionDetails{
 			runtimev1alpha1.ResourceCredentialsSecretPasswordKey: []byte(pw),
 		},
-	}
-
-	return ec, nil
+	}, nil
 }
 
 func (e *external) Update(ctx context.Context, mg resource.Managed) (resource.ExternalUpdate, error) {
-	// TODO(negz): Support updating Azure SQL servers. :)
-
-	return resource.ExternalUpdate{}, nil
+	cr, ok := mg.(*v1alpha3.MySQLServer)
+	if !ok {
+		return resource.ExternalUpdate{}, errors.New(errNotMySQLServer)
+	}
+	// TODO(muvaf): If an async update operation is ongoing, state is still ready
```

Comment:
We need to check with the `Future` object whether the operation is still ongoing as received `mysql.Server` object is misleading. But we don't have access to that `Future` object that is created by the `Update` call in the next reconciles. I've tried a few things but not much success so far. Basically, we're dealing two different source of truth for the state and one is really hard to know.

Reply:
Here is the current update flow:
- We call `Update` when there is a drift. It gets accepted and operation starts. We only check once whether operation is going through, right after the `Update` call.
- Updated parameters are immediately reflected on the received object even though the operation is still ongoing. For example, I've increased the storage from 20GB to 25GB and the first received object already had `25GB`. But in the console UI, you still see `20GB`. Only after the operation is completed, `25GB` is reflected on the UI.

I guess console UI uses a different API because the object returned from `Get` call is different than what UI shows. However, I was suspecting that we'll call `Update` call continuously since we can't really know if an operation is ongoing but that doesn't happen since we compare with what we receive from `Get` and that's immediately updated.

It works but the caveat is that we are not able to tell the user that an operation is ongoing and the values in spec might not be the most up-to-date values. Normally, we claim that you can rely to spec fields if the state is ready. But in this case, value in the spec is misleading as `userVisibleState` is always `Ready` even though an operation is still in progress.